src/GraphQl/Subscription/MercureSubscriptionUriGenerator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\GraphQl\Subscription;
+
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Generates Mercure-related URIs from a subscription ID.
+ *
+ * @experimental
+ *
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class MercureSubscriptionUriGenerator implements MercureSubscriptionUriGeneratorInterface
+{
+    private $requestContext;
+    private $hub;
+
+    public function __construct(RequestContext $requestContext, string $hub)
+    {
+        $this->requestContext = $requestContext;
+        $this->hub = $hub;
+    }
+
+    public function generateMercureTopicIri(string $subscriptionId): string
+    {
+        return "/subscriptions/$subscriptionId";
+    }
+
+    public function generateMercureTopicUrl(string $subscriptionId): string
+    {
+        if (('' === $scheme = $this->requestContext->getScheme()) || ('' === $host = $this->requestContext->getHost())) {
+            throw new \RuntimeException('Cannot generate topic URL from request context.');
```

Comment:
I suggest to use a default value if the context isn't available, such as `https://api-platform.com/subsriptions/`. IRIs don't have to exist.

Reply:
OK, it's an edge case anyway.

Reply:
Actually, maybe should we always use this default, and let the user pass another namespace if required, instead of relying on the router.